src/components/common/needsShutdown.jsx

```diff
@@ -0,0 +1,161 @@
+            (vm.cpu.threads !== vm.inactiveXML.cpu.threads) ||
+            (vm.cpu.cores !== vm.inactiveXML.cpu.cores));
+}
+
+export function needsShutdownCpuModel(vm) {
+    /* The live xml shows what host-model expanded to when started
+     * This is important since the expansion varies depending on the host and so needs to be tracked across migration
+     */
+    if (vm.inactiveXML.cpu.mode === 'host-model')
+        return !(vm.cpu.mode == 'host-model' || vm.cpu.model === vm.capabilities.cpuHostModel);
+    if (vm.inactiveXML.cpu.mode === 'host-passthrough')
+        return vm.cpu.mode !== 'host-passthrough';
+    if (vm.inactiveXML.cpu.mode === 'custom')
+        return vm.cpu.mode !== 'custom' || vm.cpu.model !== vm.inactiveXML.cpu.model;
+
+    return false;
+}
+
+export function needsShutdownWatchdog(vm) {
+    return vm.persistent && vm.state === "running" && vm.inactiveXML.watchdog.action !== vm.watchdog.action;
+}
+
+export function getDevicesRequiringShutdown(vm) {
+    if (!vm.persistent)
+        return [];
+
+    const devices = [];
+
+    // DISKS
+    for (const target in vm.disks) {
+        if (needsShutdownDiskAccess(vm, target)) {
+            devices.push(_("Disk"));
+            break;
+        }
+    }
+
+    // INTERFACES
+    for (const iface of vm.interfaces) {
+        if (needsShutdownIfaceType(vm, iface) ||
+            needsShutdownIfaceModel(vm, iface) ||
+            needsShutdownIfaceSource(vm, iface)) {
+            devices.push(_("Network interface"));
+            break;
+        }
+    }
+
+    // VCPU
+    if (needsShutdownVcpu(vm))
+        devices.push(_("vCPUs"));
+
+    // CPU
+    if (needsShutdownCpuModel(vm))
+        devices.push(_("CPU"));
+
+    // Watchdog
+    if (needsShutdownWatchdog(vm))
+        devices.push(_("Watchdog"));
+
+    return devices;
+}
+
+export const NeedsShutdownTooltip = ({ iconId, tooltipId }) => {
+    return (
+        <Tooltip id={tooltipId} content={NEEDS_SHUTDOWN_MESSAGE}>
+            <PendingIcon color="orange" id={iconId} />
+        </Tooltip>
+    );
+};
+
+export const NeedsShutdownAlert = ({ idPrefix }) =>
+    <Alert isInline variant='warning' id={`${idPrefix}-idle-message`} title={NEEDS_SHUTDOWN_MESSAGE} />;
+
+export const VmNeedsShutdown = ({ vm }) => {
+    const devices = getDevicesRequiringShutdown(vm);
+
+    if (devices.length === 0)
+        return;
+
+    const body = (
+        <>
+            <List isPlain isBordered>
+                { devices.map(device => {
+                    return (
+                        <ListItem key={device}>
+                            {cockpit.format(_("$0 configuration was changed"), device)}
+                        </ListItem>
+                    );
+                }) }
+            </List>
+        </>
+    );
+
+    const header = _("VM needs shutdown");
+    return (
+        <Popover aria-label={header}
+            alertSeverityVariant="warning"
```

Comment:
Fixed.

Reply:
Unfix.
I just realized we use orange icons (warning variant) everywhere for letting user know that VM needs reboo.
We need to be consistent. So do you think we should switch it everywhere else to info variant?

![Screenshot from 2023-04-03 12-10-10](https://user-images.githubusercontent.com/42733240/229480134-d5ec32a9-e40a-47ca-b24c-43fae0f20525.png)

Reply:
Anyway, this is what it would look like as an info variant:
@garrett what do you think?

![Screenshot from 2023-04-03 12-17-40](https://user-images.githubusercontent.com/42733240/229481797-b0db58a7-2207-496f-8d4f-85b1c9220816.png)
![Screenshot from 2023-04-03 12-17-29](https://user-images.githubusercontent.com/42733240/229481819-2e8460a9-161f-4dae-9b3f-678d18566d12.png)

Reply:
Agreed.

Reply:
Ok, updated the PR to info variant